homeassistant/components/sensor/monzo.py

```diff
@@ -0,0 +1,157 @@
+"""
+Support for Monzo service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.monzo/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_MONITORED_VARIABLES, CONF_ACCESS_TOKEN)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['monzo==0.5.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ACCOUNT_ID = 'account_id'
+
+ICON_EUR = 'mdi:currency-eur'
+ICON_GBP = 'mdi:currency-gbp'
+ICON_INR = 'mdi:currency-inr'
+ICON_NGN = 'mdi:currency-ngn'
+ICON_RUB = 'mdi:currency-rub'
+ICON_TRY = 'mdi:currency-try'
+ICON_USD = 'mdi:currency-usd'
+
+DEFAULT_NAME = 'Monzo'
+
+VARIABLE_TYPES = {
+    'balance': 'Balance',
+    'currency': 'Currency',
+    'spend_today': 'Spend today',
+    'local_currency': 'Local currency',
+    'local_exchange_rate': 'Local exchange rate'
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Required(CONF_MONITORED_VARIABLES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(VARIABLE_TYPES)]),
+    vol.Optional(CONF_ACCOUNT_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Monzo sensor."""
+    access_token = config.get(CONF_ACCESS_TOKEN)
+    account_id = config.get(CONF_ACCOUNT_ID)
+    name = config.get(CONF_NAME)
+
+    data = MonzoData(access_token, account_id)
```

Comment:
Would be nice if we could verify that the credentials are valid. Otherwise the setup of the platform should fail.